test/phoenix/router/resource_test.exs

```diff
@@ -0,0 +1,215 @@
+defmodule Phoenix.Router.ResourceTest do
+  use ExUnit.Case, async: true
+  use RouterHelper
+
+  defmodule AccountController do
+    use Phoenix.Controller
+    plug :action
+    def show(conn, _params), do: text(conn, "show account")
+    def new(conn, _params), do: text(conn, "new account")
+    def edit(conn, _params), do: text(conn, "edit account")
+    def create(conn, _params), do: text(conn, "create account")
+    def update(conn, _params), do: text(conn, "update account")
+    def delete(conn, _params), do: text(conn, "delete account")
+  end
+
+  defmodule Api.SessionController do
+    use Phoenix.Controller
+    plug :action
+    def show(conn, _params), do: text(conn, "show session")
+    def new(conn, _params), do: text(conn, "new session")
+  end
+
+  defmodule Api.CommentController do
+    use Phoenix.Controller
+    plug :action
+    def index(conn, _params), do: text(conn, "index comments")
+    def show(conn, _params), do: text(conn, "show comments")
+    def new(conn, _params), do: text(conn, "new comments")
+    def create(conn, _params), do: text(conn, "create comments")
+    def update(conn, _params), do: text(conn, "update comments")
+    def delete(conn, _params), do: text(conn, "delete comments")
+    def special(conn, _params), do: text(conn, "special comments")
+  end
+
+  defmodule Router do
+    use Phoenix.Router
+
+    resource "/account", AccountController, alias: Api do
+      resources "/comments", CommentController do
+        get "/special", CommentController, :special
+      end
+      resource "/session", SessionController, except: [:delete]
+    end
+
+    resource "/session", Api.SessionController, only: [:show]
```

Comment:
We can skip those routes and the ones below and their respective tests IMO. Just keeping it simple. :)